test/connectors/PayApiConnectorSpec.scala

```diff
@@ -73,4 +75,88 @@ class PayApiConnectorSpec extends BaseSpec with MockitoSugar with ScalaFutures {
       }
     }
   }
+
+  "findPayments" should {
+
+    "parse the json load for a successful OK response" in {
+
+      val json = Json.parse("""
+                              |{
+                              |   "searchScope":"PTA",
+                              |   "searchTag":"1097172564",
+                              |   "payments":[
+                              |      {
+                              |         "id":"5ddbd2847a0000c7f0d845a4",
+                              |         "reference":"1097172564K",
+                              |         "amountInPence":10623,
+                              |         "status":"Successful",
+                              |         "createdOn":"2019-11-25T13:09:24.188",
+                              |         "taxType":"self-assessment"
+                              |      }
+                              |   ]
+                              |}
+                              |
+                              |""".stripMargin)
+
+      when(http.GET[HttpResponse](any())(any(), any(), any()))
+        .thenReturn(Future.successful(HttpResponse(OK, Some(json))))
+
+      connector.findPayments("111111111").futureValue shouldBe Some(json.as[PaymentSearchResult])
+    }
+
+    "parse json payload for multiple payments" in {
+
+      val json = Json.parse("""
+                              |{
+                              |   "searchScope":"PTA",
+                              |   "searchTag":"1097172564",
+                              |   "payments":[
+                              |      {
+                              |         "id":"5ddbd2847a0000c7f0d845a4",
+                              |         "reference":"1097172564K",
+                              |         "amountInPence":10623,
+                              |         "status":"Successful",
+                              |         "createdOn":"2019-11-25T13:09:24.188",
+                              |         "taxType":"self-assessment"
+                              |      },
+                              |      {
+                              |         "id":"5ddbd38f7a0000c7f0d845a5",
+                              |         "reference":"1097172564K",
+                              |         "amountInPence":20000,
+                              |         "status":"Successful",
+                              |         "createdOn":"2019-11-25T13:13:51.755",
+                              |         "taxType":"self-assessment"
+                              |      }
+                              |   ]
+                              |}
+                              |
+                              |""".stripMargin)
+
+      when(http.GET[HttpResponse](any())(any(), any(), any()))
+        .thenReturn(Future.successful(HttpResponse(OK, Some(json))))
+
+      connector.findPayments("111111111").futureValue shouldBe Some(json.as[PaymentSearchResult])
+    }
+
+    "Returns a None when the status code is not OK" in {
```

Comment:
Because Wiremock is not being used in this connector Spec. This test isn't validly testing the responses from the endpoint. We are making assumptions within these test that may not actually be the case in live. For example the response in this particular test wouldn't be a BadRequest. It would be an Upstream4xxException.